sdk/core/Azure.Core/src/ResponseClassifier.cs

```diff
@@ -56,8 +60,51 @@ public virtual bool IsRetriable(HttpMessage message, Exception exception)
         /// </summary>
         public virtual bool IsErrorResponse(HttpMessage message)
         {
+            if (CustomError(message.Response.Status, out bool isError))
+            {
+                return isError;
+            }
+
             var statusKind = message.Response.Status / 100;
             return statusKind == 4 || statusKind == 5;
         }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="statusCode"></param>
+        /// <param name="isError"></param>
+        /// <returns></returns>
+        protected bool CustomError(int statusCode, out bool isError)
+        {
+            if (_customErrors?.Contains(statusCode) ?? false)
+            {
+                isError = true;
+                return true;
+            }
+
+            if (_customNonErrors?.Contains(statusCode) ?? false)
+            {
+                isError = false;
+                return true;
+            }
+
+            isError = false;
+            return false;
+        }
+
+        internal void CustomizeErrors(int[]? customErrors, int[]? customNonErrors)
+        {
+            if (customErrors != null)
+            {
+                _customErrors = new int[customErrors.Length];
+                Array.Copy(customErrors, _customErrors, customErrors.Length);
```

Comment:
Is it bad that we're copying these values twice?  (Once in ConfigureResponse on the RequestContext, once here to customize the classifier)

Reply:
Maybe we can make classifiers immutable?